src/App.js

```diff
@@ -1,16 +1,45 @@
 import React from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import { useState } from 'react';
+import ChatLog from './components/ChatLog.js';
 
 const App = () => {
+  const [chatData, setChatData] = useState(chatMessages);
+
+  const likedMessages = (id) => {
+    const newLikedData = chatData.map((chat) => {
+      if (chat.id === id) {
+        return { ...chat, liked: (chat.liked = !chat.liked) };
+      } else {
+        return chat;
+      }
+    });
+
+    setChatData(newLikedData);
+  };
+
+  const displayLikes = () => {
+    const likedData = chatData.filter((chat) => {
+      if (chat.liked) {
+        return chat;
+      }
+    });
+    let hearts = '';
+    likedData.forEach((like) => {
+      hearts += '❤️';
+    });
+    return hearts;
```

Comment:
We really want to be returning the count of hearts rather than a string made up of the number of hearts, so something like the following:
```js
    let hearts = 0;
    likedData.forEach((like) => {
      ++hearts;
    });
    return hearts;
```

But if we _did_ want to build a string of the hearts, remember that like Python, JS strings are immutable, so repeated concatenation tends towards quadratic time. Instead, we could do something (assuming `likedData` is filtered as above) like
```js
    return likedData.map(_ => '❤️').join('');
```

We use `_` for the parameter to acknowledge that a value is passed in, but we're not using it. But we could just as easily do
```js
    return likedData.map(() => '❤️').join('');
```
Since JS doesn't enforce that the number of arguments and parameters match.